### PR TITLE
Performance audit

### PR DESCRIPTION
**Pull request by emmanuelsteven**
In this PR ;
- I altered the owner's table and added an email column.

- Added data to the visits' and owners' database

- Used EXPLAIN ANALYZE on the queries to ensure the estimated execution time was above 1000 ms on the following :
- SELECT COUNT(*) FROM visits where animal_id = 4; 
animals
 - SELECT * FROM visits where vet_id = 2; 
Screenshot 2023-07-05 at 18 14 23
- SELECT * FROM owners where email = 'owner_18327@mail.com';
email
Optimized
I found a way to improve the execution time of the above queries using create index. Results below:
CREATE INDEX idx_animal_id ON visits (animal_id);
reduce
CREATE INDEX idx_vet_id ON visits (vet_id);
Screenshot 2023-07-05 at 18 12 17
CREATE INDEX  idx_email ON owners(email);
emailsmall

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -57,9 +57,10 @@ CREATE TABLE specializations (
...
-    vet_id INT,
     animal_id INT,
+    vet_id INT,
     visit_date DATE,
-    FOREIGN KEY (vet_id) REFERENCES vets(id),
-    FOREIGN KEY (animal_id) REFERENCES animals(id) 
+    PRIMARY KEY (animal_id, vet_id, visit_date),
+    FOREIGN KEY (animal_id) REFERENCES animals (id),
+    FOREIGN KEY (vet_id) REFERENCES vets (id)
 );
```
Comment (mikethreels) [inline schema.sql]: - [ ] good job making the performance better, however, adding indexes for FOREIGN KEYS or on columns frequently used in queries is one of the basic solutions for the performance problems please try it out before requesting another review.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Emmanuel, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- good job improving the queries.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema.sql (hunk 1 above)
Comment (Munsa1) [reply schema.sql]: - [ ] As requested by the previous reviewer, kindly add indexes for FOREIGN KEYS on columns frequently used for the queries in your `schema.sql` for code documentation purposes. 

The following is just an example, your implementation can be different.
e.g.
``` 
   CREATE INDEX ON visits (animal_id);
   CREATE INDEX ON visits (vet_id);
   CREATE INDEX ON owners (email);
```

### Note that your screenshots are okay you just need the code in `schema.sql`

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @emmanuelsteven, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
Passing Linters  N/A
Correct Git Flow :heavy_check_mark: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

- [ ] Since you have not added anything to the `FAQ` section of your readme file, consider removing it altogether from your readme and from its table of contents
image

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Felix45 — APPROVED** [review]
# Approved :green_circle: 
Hi @emmanuelsteven, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights :heavy_check_mark: 
- [x] Good SQL query design :white_check_mark: 
- [x] Indexes added to all foreign keys :white_check_mark: 
- [x] Implemented all the changes from the previous reviewers :white_check_mark:  

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**